graphdatascience/graph/nx_loader.py

```diff
@@ -0,0 +1,124 @@
+from collections import defaultdict
+from typing import Any, Dict, List, Set, Tuple
+
+import networkx as nx
+from pandas import DataFrame
+
+from ..error.client_only_endpoint import client_only_endpoint
+from ..error.illegal_attr_checker import IllegalAttrChecker
+from ..error.uncallable_namespace import UncallableNamespace
+from ..server_version.compatible_with import compatible_with
+from ..server_version.server_version import ServerVersion
+from .graph_object import Graph
+
+
+class NXLoader(UncallableNamespace, IllegalAttrChecker):
+    @client_only_endpoint("gds.graph.networkx")
+    @compatible_with("load", min_inclusive=ServerVersion(2, 1, 0))
+    def load(self, nx_G: nx.Graph, graph_name: str, concurrency: int = 4) -> Graph:
+        nodes, rels = self._parse(nx_G)
+
+        undirected_rel_types = []
+        if not isinstance(nx_G, nx.DiGraph) and not isinstance(nx_G, nx.MultiDiGraph):
+            undirected_rel_types = [df["relationshipType"][0] for df in rels]
+
+        constructor = self._query_runner.create_graph_constructor(graph_name, concurrency, undirected_rel_types)
+        constructor.run(nodes, rels)
+
+        return Graph(graph_name, self._query_runner, self._server_version)
+
+    @staticmethod
+    def _parse(nx_G: nx.Graph) -> Tuple[List[DataFrame], List[DataFrame]]:
+        node_dicts_by_labels: Dict[Tuple[str, ...], Dict[str, List[Any]]] = defaultdict(lambda: defaultdict(list))
+        node_props_schema: Dict[Tuple[str, ...], Set[str]] = {}
+        no_node_labels = None
+
+        for node_id, attrs in nx_G.nodes(data=True):
+            labels_attr = attrs.get("labels", None)
+            if no_node_labels is None:
+                no_node_labels = labels_attr in [None, []]
+
+            if (labels_attr in [None, []]) is not no_node_labels:
```

Comment:
Why do we have this limitation couldnt we handle this in the projections? 
Just checked and you can use `[] or null`

Reply:
I get errors if I try to load with node labels `[]` or `null` using Arrow as the backend

Reply:
ah .. not sure if its expected :open_mouth:

Reply:
https://trello.com/c/gv7FxQO2/8047-allow-empty-node-labels-when-projecting-through-arrow created a card

Reply:
thank you! :pray:

Reply:
Keeping it as then for now